sabnzbd/downloader.py

```diff
@@ -1014,3 +1018,29 @@ def clues_pay(text: str) -> bool:
         if clue in text:
             return True
     return False
+
+
+def check_server_expiration():
+    """Check if user should get warning about server date expiration"""
+    for server in config.get_servers().values():
+        if server.expire_date():
+            days_to_expire = ceil(
+                (time.mktime(time.strptime(server.expire_date(), "%Y-%m-%d")) - time.time()) / (60 * 60 * 24)
+            )
+            # Notify from 5 days in advance
+            if 0 <= days_to_expire < 6:
```

Comment:
I would skip the 0 <= so there is at least one warning, even if SAB is only started after it has expired. I think this feature is particularly useful for free xsusenet, which most users probably don't use as a primary server. There won't necessarily be a warning from the downloader for a while.

Reply:
Done